docs/ref/models/constraints.txt

```
@@ -34,6 +34,17 @@ option.
     Constraints are checked during the :ref:`model validation
     <validating-objects>`.
 
+.. admonition:: Validation of Constraints with ``JSONField``
+
+    Constraints containing :class:`~django.db.models.JSONField` may not raise
+    validation errors as key, index, and path transforms have many
+    database-specific caveats. This :ticket:`may be fully supported later
+    <34059>`.
+
+    You should always check log messages, in the ``django.db.models`` logger,
+    looking for no messages like *"Got a database error calling check() on …"*
+    to confirm it's validated properly.
```

`…looking for no message like…` feels a little awkward. Maybe... 

```suggestion
    You should always check that there are no log messages, in the
    ``django.db.models`` logger, like *"Got a database error calling check() on
    …"* to confirm it's validated properly.
```

---

Thanks!